src/main/java/org/spongepowered/api/text/Text.java

```diff
@@ -107,6 +130,8 @@
         this.clickAction = Optional.<ClickAction<?>>ofNullable(clickAction);
         this.hoverAction = Optional.<HoverAction<?>>ofNullable(hoverAction);
         this.shiftClickAction = Optional.<ShiftClickAction<?>>ofNullable(shiftClickAction);
+        this.childrenIterable = () ->
+                this.children.isEmpty() ? Iterators.singletonIterator(this) : new TextIterator(this);
```

Comment:
Your first example has no difference to what it is currently (keep in mind it is a lambda, so `isEmpty()` would still be checked every time), except that it causes a compile error.
I may change it like in your second example, however not sure if that is really worth it...